### PR TITLE
Fix issue with parsing OPW sign correction from SRDF.

### PR DESCRIPTION
**Pull request by Levi-Armstrong**
This PR also adds some error checking for Joint and State waypoint to make sure joint names and positions are the same length.

### CODE REVIEW

File: tesseract/tesseract_scene_graph/test/tesseract_scene_graph_unit.cpp [hunk 1]
```diff
@@ -347,6 +348,42 @@ TEST(TesseractSceneGraphUnit, LoadSRDFUnit)  // NOLINT
...
+  EXPECT_NEAR(opw.offsets[4], 0.0, 1e-8);
+  EXPECT_NEAR(opw.offsets[5], 0.0, 1e-8);
+  EXPECT_EQ(opw.sign_corrections[0], 1);
+  EXPECT_EQ(opw.sign_corrections[1], 1);
+  EXPECT_EQ(opw.sign_corrections[2], 1);
+  EXPECT_EQ(opw.sign_corrections[3], -1);
+  EXPECT_EQ(opw.sign_corrections[4], 1);
+  EXPECT_EQ(opw.sign_corrections[5], 1);
+}
+
```
Comment (mpowelson) [inline tesseract/tesseract_scene_graph/test/tesseract_scene_graph_unit.cpp:386]: This test doesn't test when the OPW sign correction is wrong does it?

Comment (Levi-Armstrong) [reply tesseract/tesseract_scene_graph/test/tesseract_scene_graph_unit.cpp:386]: No, it is testing the issue with the parser of the xml. I was casting to a signed char which was not working if -1. So before this change the test would not pass but now it should.

Comment (Levi-Armstrong) [reply tesseract/tesseract_scene_graph/test/tesseract_scene_graph_unit.cpp]: I need to go through and create full set of unit tests for the SRDF similar to the URDF.

Comment (mpowelson) [reply tesseract/tesseract_scene_graph/test/tesseract_scene_graph_unit.cpp]: If you want to merge this before doing that, that is fine with me.

Comment (Levi-Armstrong) [reply tesseract/tesseract_scene_graph/test/tesseract_scene_graph_unit.cpp]: Lets wait. I should have time over the next month to go through and add these.

**Comment by Levi-Armstrong** [comment]
@mpowelson Are you good with merging this one?

**Comment by Levi-Armstrong** [comment]
Included in PR #453

**Pull request closed without merging**